### PR TITLE
fixed sorting bug in maturity-index

### PR DESCRIPTION
**Pull request by nbokulich**
scikit-learn evidently uses feature order — NOT feature labels (!!!!) — for model training. Hence, if these features are sorted in a different order in the test dataset, the resulting predictions make no sense.

This only impacted `maturity-index` because an additional set of predictions is performed — on the full sample set, which must be sorted in the same order as the training data. The results were quite fatal — see the difference between old and new example plots below.

Also fixed deprecated use of `.ix` and `maturity-index` parameter summaries that were not visualizing correctly.

### CODE REVIEW

**Comment by nbokulich** [comment]
thanks @gregcaporaso ! I just added a test for this in the most recent commit.
